broadinstitute/pearl
# [JN-406] postgres search prototype

This adds proof-of-concept filtering of the participant list via Postgres.  
![image](https://github.com/broadinstitute/juniper/assets/2800795/0b67aa68-563b-474a-ad5f-0a8e0b0183cf)
I've walked Nick through the code generally, and I'll do the same with Matt.  I'm not sure the code deserves too much attention, since we'll be refining this A LOT as we add non-hardcoded facets and saved filters and whatnot.  What we should be paying the most attention to is whether all this code seems maintainable, with an eye towards when/if/how we should use elasticSearch or some other search cache instead (or likewise use a library for the facet UI).  the big advantage of this approach is it doesn't require a separate cache/infrastructure.  Which is HUGE, because it makes troubleshooting WAY easier, and we don't have to worry about sync.   the flip side, is that we have some pretty gnarly sql mapping instead.  

TO REVIEW:
1. Start with looking at enrolleeSearch.ts and the SAMPLE_FACETS -- that gives you the data structure for the facets themselves. 
2. Look at facetsToString and facetsFromString to get a sense of how we are mapping these to URL params
3. Look at EnrolleeSearchDaoTests to see the filters and how they're built on the Java side, and how they're tested.
4. Skim the rest -- it will probably be either substantially upgraded as we move this beyond POC, or scrapped in facvor of a totally different approach

TO TEST:
1. reload adminApiApp
2. go to https://localhost:3000/ourhealth/studies/ourheart/env/sandbox/participants
3. futz around with the filters, confirm they work
4. reload the page with various filters specified, confirm they persist
